src/poolCaching/onchainData.ts

```diff
@@ -94,19 +87,48 @@ export async function getOnChainBalances(
         }
     >;
 
-    Object.entries(pools).forEach(([poolId, onchainData]) => {
+    try {
+        pools = (await multiPool.execute()) as Record<
```

Comment:
Ideally we'd fix the return value from `multipool.execute` to be compatible with the `Record`. I'm happy to to the type wrangling in a separate PR though.

Reply:
I've left this for a future PR.